src/gtfs-rt/client.js

```diff
@@ -0,0 +1,72 @@
+import GtfsRealtimeBindings from 'gtfs-realtime-bindings';
+import logger from '../lib/logger.js';
+
+const API_KEY = process.env.GTFS_REALTIME_TOKEN;
+const ALERTS_URL = process.env.GTFS_REALTIME_ALERTS_URL;
+
+let _alertsCache;
+let _alertsCacheTime;
+let _alertsPromise;
+// We're allowed to make 60 requests per hour, let's be conservative and
+// cache for 3min to make it a maximum of 20
+const CACHE_LENGTH = 3 * 60 * 1000;
+
+export async function getAlerts() {
+  let alerts;
+  if (_alertsCache && Date.now() - _alertsCacheTime < CACHE_LENGTH) {
+    return _alertsCache;
+  }
+
+  if (!API_KEY) {
+    logger.warn('no API key defined, cannot fetch alerts');
+    return null;
+  }
+
+  if (!ALERTS_URL) {
+    logger.warn('no GTFS-RT alerts URL defined, cannot fetch alerts');
+    return null;
+  }
+
+  // If there's already a fetch in progress, we'll wait for that one.
+  // But if not, let's initiate one.
+  if (!_alertsPromise)
```

Comment:
this will not have the effect you are describing because this file isn't a singleton. each time it is imported it will create a new `_alertsPromise` for that context. assuming this file is only ever used  in one import statement then it will work as described but this comment/guard suggests it provides a level of protection that it really doesn't. I think the idea though, sadly without a DI system producing singletons is kind of a pain in the ass in JS. maybe we could just be more explicit in the comment about the depth of this gaurd?

Reply:
Hmm I don't think this is a problem in practice or else using `instanceof` within Node.js wouldn't work.

[This Medium post from 2016](https://medium.com/@lazlojuly/are-node-js-modules-singletons-764ae97519af) suggests that while there are some cases in which (old) Node.js would not cache (CommonJS) modules, we would be unlikely to hit those cases in practice.

[Current Node's documentation on ES modules](https://nodejs.org/api/esm.html) is unclear on the point.

[Here's a free ebook chapter](https://exploringjs.com/es6/ch_modules.html#sec_modules-in-javascript) that says of ES modules, "Modules are singletons. Even if a module is imported multiple times, only a single “instance” of it exists." (Found via [this StackOverflow answer](https://stackoverflow.com/questions/64477263/use-module-as-singleton), which makes the point about `instanceof`.)